### ISSUE TITLE
Windows .usd file creation on network drives can fail when preceded by specific file access checks

### ISSUE THREAD

**Issue by marktucker** [PixarAnimationStudios/USD#1870]
### Description of Issue
Windows file existence checking has the possibility to leave an error set, that can be erroneously flagged as an error in a subsequent file existence/permission check. For some reaosn this seems to only affet network drives, and only very specific sequences of file checks triggers this problem.

### Steps to Reproduce
1. On a Windows machine, where the Z: drive is a network mount, and z:/bugs exists as a directory, run the following code:
```
from pxr import Sdf
l1 = Sdf.Layer.CreateAnonymous()
l2 = Sdf.Layer.Find("z:/bugs/12345/foo/bar/thing.usd")
l1.Export("z:/bugs/12345/foo/bar/thing.usd")
```
2. The Export call will fail with the following exception:
Traceback (most recent call last):
```
  File "<console>", line 1, in <module>
pxr.Tf.ErrorException: 
        Error in 'pxrInternal_v0_22__pxrReserved__::TfSafeOutputFile::Replace' at line 138 
in file F:\dev_tools\src\usd\usd-22.05\USD-py3.9\pxr\base\tf\safeOutputFile.cpp : 'U
nable to determine the real path for 'z:/bugs/121495/foo/bar/thing.usd': The system 
cannot find the path specified.

'Error in 'pxrInternal_v0_22__pxrReserved__::Usd_CrateFile::CrateFile::StartPacking'
 at line 2506 in file F:\dev_tools\src\usd\usd-22.05\USD-py3.9\pxr\usd\usd\crateFile
.cpp : 'Unable to open z:/bugs/121495/foo/bar/thing.usd for write'
```

### System Information (OS, Hardware)
Windows only.

**Comment by jilliene** [comment]
Filed as internal issue #USD-7395

**Comment by sunyab** [comment]
This fix was released in v22.08. Closing, thanks!
